backend/src/question/question.service.ts

```diff
@@ -9,32 +9,82 @@ import {
   QUESTION_SUMMARY_SELECT,
 } from "./question.type";
 import { PrismaService } from "../prisma/prisma.service";
+import { NAMESPACES } from "shared/api";
+import { RedisCacheService } from "src/cache/redisCache.service";
+import {
+  QUESTION_CONTENT,
+  QUESTION_QOTD_SUMMARY,
+  QUESTION_QOTD_CONTENT,
+  QUESTION_SUMMARIES,
+  QUESTION_TOPICS,
+  QUESTION_SUMMARY_DIFFICULTIES,
+  QUESTION_SUMMARIES_FORMATTED,
+} from "./question.cache.keys";
+import { Cron } from "@nestjs/schedule";
 
 @Injectable()
 export class QuestionService {
-  constructor(private prisma: PrismaService) {}
+  constructor(
+    private prisma: PrismaService,
+    private cache: RedisCacheService
+  ) {}
 
+  recentlyCached: string[] = [];
   /**
    * Gets all the question summaries with the following fields:
    * acRate, difficulty, title, titleSlug, topicTags and updatedAt
    *
    * @return  Array of QuestionSummary with the relevant fields
    */
   async getAllSummaries() {
-    const res: QuestionSummaryFromDb[] =
-      await this.prisma.questionSummary.findMany({
-        select: QUESTION_SUMMARY_SELECT,
-      });
+    const summaries = await this.cache.getKeyInNamespace<
+      FlattenedQuestionSummary[]
+    >([NAMESPACES.QUESTIONS], QUESTION_SUMMARIES_FORMATTED);
+
+    if (!summaries) {
+      const res: QuestionSummaryFromDb[] =
+        await this.prisma.questionSummary.findMany({
+          select: QUESTION_SUMMARY_SELECT,
+        });
 
-    return this.formatQuestionSummaries(res);
+      const formattedQuestionSummaries = this.formatQuestionSummaries(res);
+
+      this.cache.setKeyInNamespace(
+        [NAMESPACES.QUESTIONS],
+        QUESTION_SUMMARIES,
+        res
+      );
+
+      this.cache.setKeyInNamespace(
+        [NAMESPACES.QUESTIONS],
+        QUESTION_SUMMARIES_FORMATTED,
+        formattedQuestionSummaries
+      );
```

Comment:
```suggestion
```
Not sure if I'm understanding this right -- the question summaries have already been cached so this formatted response isn't really required since it's a matter of passing the cached value into `this.formatQuestionSummaries()`. If you feel this is needed we can discuss 👍